components/selection/selection-summary.js

```diff
@@ -35,25 +36,54 @@ class Summary extends LocalizeCoreElement(SelectionObserverMixin(LitElement)) {
 	}
 
 	render() {
+		if (!this._summary) {
+			return nothing;
+		}
+		return html`
+			<p class="d2l-body-compact">${this._summary}</p>
+		`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('_provider') || changedProperties.has('selectionInfo')) {
+			this._updateSelectSummary();
+		}
+	}
+
+	_updateSelectSummary() {
 		if (this._provider && this._provider.selectionSingle) return;
 
 		let count;
-		let summary;
 		if (this._provider && this._provider.selectionCountOverride !== undefined) {
 			count = this._provider.selectionCountOverride;
-			summary = (this._provider.selectionCountOverride === 0 && this.noSelectionText ?
-				this.noSelectionText : this.localize('components.selection.selected', 'count', count));
+			this._summary = this._provider.selectionCountOverride === 0 && this.noSelectionText ?
+				this.noSelectionText : this.localize('components.selection.selected', 'count', count);
 		} else {
-			count = (this.selectionInfo.state === SelectionInfo.states.allPages ?
-				this._provider.itemCount : this.selectionInfo.keys.length);
+			let includePlus = false;
+			if (this._provider && this._provider.getItems) {
+				const lazyLoadListItems = this._provider.getItems()[0]._getFlattenedListItems().lazyLoadListItems;
```

Comment:
Note that `d2l-selection-summary` is a generic component that could be used by `d2l-list`, and also `d2l-table-wrapper`, but this method relies on a list specific method, `_getFlattenedListItems`. I think we would want to figure out how to represent this generically as to push the list-specific aspect of this to the actual `d2l-list` class.

Reply:
I could add essentially add this to the list item if that's preferred:
```
_getLazyLoadItems() {
 return this.getItems()[0]._getFlattenedListItems().lazyLoadListItems;
}
```

And then this would change to just:
```
if (this._provider && this._provider._getLazyLoadItems) {
  const lazyLoadListItems = this._provider._getLazyLoadItems();
}
```

Thoughts?

Reply:
Yeah that sounds ok. We'll want to add a default implementation of `_getLazyLoadItems` to [SelectionMixin](https://github.com/BrightspaceUI/core/blob/main/components/selection/selection-mixin.js#L38) that returns an empty map since `d2l-table-wrapper` won't have a need to implement this.

Reply:
We shouldn't need to add this to the `SelectionMixin`. I check `if (this._provider && this._provider._getLazyLoadItems)` in selection-summary so it won't execute if `_getLazyLoadItems` is undefined.